### ISSUE TITLE
When using comment types, im getting errors on all self references

### ISSUE THREAD

**Issue by cebor** [microsoft/pyright#1232]
## Environment data

Name: Python
Id: ms-python.python
Description: Linting, Debugging (multi-threaded, remote), Intellisense, Jupyter Notebooks, code formatting, refactoring, unit tests, snippets, and more.
Version: 2020.11.371526539
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-python.python

Name: Pylance
Id: ms-python.vscode-pylance
Description: A performant, feature-rich language server for Python in VS Code
Version: 2020.12.0
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance

Python3.9 on macOS 11

## Logs

```error
Error: Type of "self" could not be determined because it refers to itself Pylance (reportGeneralTypeIssues)
```

## Code Snippet / Additional information

```python
class MyClass():
  def myMethod(self, foo):
   # type: (str) -> str
     self.foo = foo
     return self.foo
```

**Comment by erictraut** [comment]
Thanks for the bug report. This was a recently-introduced regression that uncovered a hole in our test suite.

**Comment by erictraut** [comment]
A recent change exposed an older bug in the code that handles comment-style function annotations. I've fixed the bug and added more unit tests to cover this case better. The fix will be in the next release.

**Comment by erictraut** [comment]
@jakebailey @judej Just wanted to give you visibility into this issue, which is a regression in the version of pylance that was released yesterday. It has the potential to affect many users. I've checked in a fix to the pyright repo if you want to consider it for a hot fix.

**Comment by erictraut** [comment]
This is now fixed in Pyright 1.1.92, which I just published. It will also be addressed in next week's Pylance release.
